docs/design/storage-auto-scale.md

```diff
@@ -0,0 +1,95 @@
+# Storage Automatic Scaler
+
+## Summary
+
+This feature would help users to get rid of maintenance task of keeping the system with enough available free space.
+In virtualized environments, we can scale the storage devices OSDs so this feature helps in automatic scaling. 
+
+## Proposal
+
+Implement auto storage scale CR per device class of a specific Storagecluster.
+
+## High-Level Design
+
+- Create a StorageAutoScale CR per Storagecluster device class
+    ```
+    apiVersion: ocs.openshift.io/v1
+    Kind: StorageAutoScaling
+    Metadata:
+        name: <storageclusterName>-<deviceClassName>
```

Comment:
As a user facing API,  you should not dictate rules the name the user gives to the CR. I would suggest leaving the name be and have the spec point to the storage cluster and device class names

Reply:
We already have spec for it, we need a systematic unique naming of CRs so its just for that

Reply:
Am I understanding that this part of the design is intended to explain how ocs-operator or the UI will name the resource when scaling is turned on? If so, it would be worth explaining that in the design so we are clear on what the plan is.

Reply:
We want the CR names to be unique that is created by UI, so its just a UI requirement flow and user can have any names

Reply:
For clarity, how about we just change the design doc to this? If it's just a UI convention, we don't need to declare it in this design doc.
```suggestion
        name: <name>
```